### PR TITLE
TINF-423 | Fix locate-me button

### PR DESCRIPTION
**Pull request by florianredmann**
- Fixed sizes

### CODE REVIEW

File: SAKPaaS/src/app/components/locate-button/locate-button.component.scss [hunk 1]
```diff
@@ -3,7 +3,7 @@ button {
   position: absolute;
   width: 27.5px;
   height: 27.5px;
-  bottom: 40px;
+  bottom: calc(7vh + 20px);
```
Comment (nclskfm) [inline SAKPaaS/src/app/components/locate-button/locate-button.component.scss]: sieht irgendwie komisch aus ...
image

**Review by nclskfm — CHANGES_REQUESTED** [review]

**Review by nclskfm — DISMISSED** [review]
lgtm

**Review by lepathie — CHANGES_REQUESTED** [review]
This doesn't solve the problem at it's root. You should propably just make the container/overlay smaller on the y-axis.

**Comment by florianredmann** [comment]
Ist das wirklich notwendig? Ist das nicht einfach nur ein anderer Lösungsweg?

**Comment by lepathie** [comment]
Looks like you are using relative measures. The overlap of the map and the control also changes in height, propably should take a look at it there.
image

**Comment by lepathie** [comment]
image
Because you are not fixing the problem at it's root, the other overlay elements are and will be still behind the control. Creating css-rules for each overlay-element individually would be a very inefficient and non-future-proof approach.  
That's why I think it is necessary to just make the container of the overlay elements smaller.

I just noticed the app-locate-button-component is not inside the overlay-container but directly under map. Would it be an option to move it to where all other overlay-elements are?
image

File: SAKPaaS/src/app/components/locate-button/locate-button.component.scss [hunk 2]
```diff
@@ -24,10 +24,6 @@ button {
 @media (max-width: 600px) {
   button {
     right: 7.5vw;
-  }
-}
-@media (max-height: 800px) {
-  button {
-    top: 240px;
+    bottom: calc(7vh + 20px);
   }
```
Comment (lepathie) [inline SAKPaaS/src/app/components/locate-button/locate-button.component.scss:28]:
```suggestion
  }
  .map[_ngcontent-vvb-c82] {
      padding-bottom: 50px;
      height: calc(30vh + 50px);
  }
  .ol-unselectable, .ol-viewport {
      overflow: visible !important;
   }
```

File: SAKPaaS/src/app/components/locate-button/locate-button.component.scss [hunk 3]
```diff
@@ -24,10 +24,6 @@ button {
 @media (max-width: 600px) {
   button {
     right: 7.5vw;
```
Comment (lepathie) [inline SAKPaaS/src/app/components/locate-button/locate-button.component.scss]: Use `px` or something like `em` here.
```suggestion
    right: 20px;
```

File: SAKPaaS/src/app/components/locate-button/locate-button.component.scss [hunk 4]
```diff
@@ -24,10 +24,6 @@ button {
...
-  }
-}
-@media (max-height: 800px) {
-  button {
-    top: 240px;
+    bottom: calc(7vh + 20px);
```
Comment (lepathie) [inline SAKPaaS/src/app/components/locate-button/locate-button.component.scss]:
```suggestion
    bottom: calc(50px + 25px);
```

The `50px` are the bottom-padding on the map.

**Review by lepathie — CHANGES_REQUESTED** [review]
I propose some changes that should fix the problem at it's root.

File: SAKPaaS/src/app/components/locate-button/locate-button.component.scss (hunk 2 above)
Comment (lepathie) [reply SAKPaaS/src/app/components/locate-button/locate-button.component.scss:28]: `30vh` because thats the `top` of the location-panel on mobile.

Comment (lepathie) [reply SAKPaaS/src/app/components/locate-button/locate-button.component.scss:28]: Makes the map Overlap by exactly `50px` with the location-panel and hinders the overlays from doing so.

**Comment by lepathie** [comment]
You might also want to `display: none` the `.ol-attribution` for aesthetics if that isn't forbidden by the policy.

**Review by lepathie — DISMISSED** [review]
I am happy now. 🙂

**Comment by cgomm** [comment]
> 
> 
> I am happy now. 🙂

Hast du deine Änderungen für approved erklärt..?

**Comment by lepathie** [comment]
> > 
> > 
> > I am happy now. 🙂
> 
> Hast du deine Änderungen für approved erklärt..?

Yes.

**Review by patrickstecker — DISMISSED** [review]
LGTM

**Review by patrickstecker — APPROVED** [review]
STILL BALLIN

**Review by lepathie — APPROVED** [review]
Delicious! 🍯

**Pull request merged**